From 478804780d0873d53996eec0f05e04c1793530d8 Mon Sep 17 00:00:00 2001
From: Thomas Haller <thaller@redhat.com>
Date: Thu, 30 Aug 2018 17:12:20 +0200
Subject: [PATCH 10/14] libnm: fix leaking private-key in
 nm_setting_802_1x_set_phase2_private_key()

(cherry picked from commit 98ca7022e35a7dd735f02a1c7bcd2ed8113562cf)
---
 libnm-core/nm-setting-8021x.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libnm-core/nm-setting-8021x.c b/libnm-core/nm-setting-8021x.c
index 3b4240e64..482cbbe2b 100644
--- a/libnm-core/nm-setting-8021x.c
+++ b/libnm-core/nm-setting-8021x.c
@@ -2639,6 +2639,7 @@ nm_setting_802_1x_set_phase2_private_key (NMSetting8021x *setting,
 		password_changed = TRUE;
 	}
 
+	g_bytes_unref (priv->phase2_private_key);
 	if (scheme == NM_SETTING_802_1X_CK_SCHEME_BLOB) {
 		/* FIXME: potential race after verifying the private key above */
 		/* FIXME: ensure blob doesn't start with file:// */
-- 
2.17.0

